pallets/ajuna-affiliate/src/tests.rs

```diff
@@ -0,0 +1,302 @@
+	fn affiliate_to_rejects_if_account_is_affiliator() {
+		ExtBuilder::default().build().execute_with(|| {
+			let state = AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 0 };
+			Affiliators::<Test>::insert(BOB, state);
+			Affiliators::<Test>::insert(ALICE, state);
+
+			assert_ok!(<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+				&BOB, &ALICE
+			));
+
+			assert_noop!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+					&ALICE, &BOB
+				),
+				Error::<Test>::CannotAffiliateExistingAffiliator
+			);
+		});
+	}
+
+	#[test]
+	fn affiliate_to_rejects_affiliating_to_more_than_one_account() {
+		ExtBuilder::default().build().execute_with(|| {
+			let state = AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 0 };
+			Affiliators::<Test>::insert(BOB, state);
+			Affiliators::<Test>::insert(CHARLIE, state);
+
+			assert_ok!(<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+				&BOB, &ALICE
+			));
+
+			assert_noop!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+					&CHARLIE, &ALICE
+				),
+				Error::<Test>::CannotAffiliateAlreadyAffiliatedAccount
+			);
+		});
+	}
+
+	#[test]
+	fn affiliate_to_rejects_with_unaffiliable_account() {
+		ExtBuilder::default().build().execute_with(|| {
+			assert_noop!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+					&BOB, &ALICE
+				),
+				Error::<Test>::CannotAffiliateToAccount
+			);
+		});
+	}
+
+	#[test]
+	fn affiliate_to_rejects_with_blocked_account() {
+		ExtBuilder::default().build().execute_with(|| {
+			Affiliators::<Test>::mutate(BOB, |state| {
+				state.status = AffiliatableStatus::Blocked;
+			});
+
+			assert_noop!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+					&BOB, &ALICE
+				),
+				Error::<Test>::CannotAffiliateToAccount
+			);
+		});
+	}
+}
+
+mod clear_affiliation {
+	use super::*;
+
+	#[test]
+	fn clear_affiliation_should_work() {
+		ExtBuilder::default().balances(&[(ALICE, 1_000_000)]).build().execute_with(|| {
+			let state = AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 0 };
+			Affiliators::<Test>::insert(BOB, state);
+
+			assert_ok!(<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+				&BOB, &ALICE
+			));
+
+			System::assert_last_event(mock::RuntimeEvent::Affiliates(
+				crate::Event::AccountAffiliated { account: ALICE, to: BOB },
+			));
+
+			assert_eq!(Affiliators::<Test>::get(BOB).affiliates, 1);
+			assert_eq!(Affiliatees::<Test>::get(ALICE), Some(bounded_vec![BOB]));
+
+			assert_ok!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_clear_affiliation_for(
+					&ALICE
+				)
+			);
+
+			assert_eq!(Affiliators::<Test>::get(BOB).affiliates, 0);
+			assert_eq!(Affiliatees::<Test>::get(ALICE), None);
+		});
+	}
+
+	#[test]
+	fn clear_affiliation_if_no_affiliation_exists() {
```

Comment:
Add the expected outcome to the name: `clear_affiliation_returns_ok_if_no_affiliation_exists` or maybe we can even use the frame supports `assert_noop!` macro?